src/storage/token_buffer.rs

```diff
@@ -62,6 +97,35 @@ impl TokenBuffer {
             token,
             buffer,
             written: 0,
+            uncommitted_pages: vec![],
```

Comment:
Here you're adding a vector containing megabyte sized elements to a small wrapper type, making it very heavy. I think a design requiring this might have to be re-thought.